cluster-autoscaler/utils/test/test_utils.go

```diff
@@ -64,6 +65,19 @@ func BuildTestPod(name string, cpu int64, mem int64) *apiv1.Pod {
 	return pod
 }
 
+// RequestGpuForPod modifies pod's resource requests by adding a number of GPUs to them.
+func RequestGpuForPod(pod *apiv1.Pod, gpusCount int64) {
```

Comment:
You're adding a taint automatically in AddGpusToNode. Shouldn't you add toleration here in similar way? How is it even working without toleration?

Reply:
True. Actually we are not using the same predicates set in this UT as one which is used in production (`simulator.NewTestPredicateChecker` vs `simulator.NewPredicateChecker`).

This is poor but not much we can do about it know as we discussed offline.